lxd/container.go

```diff
@@ -748,22 +748,24 @@ func containerCreateInternal(d *Daemon, args containerArgs) (container, error) {
 
 func containerConfigureInternal(c container) error {
 	// Find the root device
-	for _, m := range c.ExpandedDevices() {
-		if m["type"] != "disk" || m["path"] != "/" || m["size"] == "" {
-			continue
-		}
+	localDevices := c.LocalDevices()
+	_, rootDiskDevice := containerGetRootDiskDevice(localDevices)
+	if rootDiskDevice["pool"] == "" {
+		expandedDevices := c.ExpandedDevices()
+		_, rootDiskDevice = containerGetRootDiskDevice(expandedDevices)
+	}
```

Comment:
ah, I think you misunderstood. I didn't meant to just change that one line. I meant you can replace the whole hunk with just that line since c.ExpandedDevices() will always return the local entry if there's one.